path/a_star.go

```diff
@@ -8,39 +8,34 @@ import (
 	"container/heap"
 
 	"github.com/gonum/graph"
+	"github.com/gonum/graph/internal"
 )
 
 // Heuristic returns an estimate of the cost of travelling between two nodes.
 type Heuristic func(x, y graph.Node) float64
 
-// HeuristicCoster wraps the HeuristicCost method. A graph that implementing
-// the interface provides a heuristic between any two given nodes.
+// HeuristicCoster wraps the HeuristicCost method. A graph implementing the
+// interface provides a heuristic between any two given nodes.
 type HeuristicCoster interface {
 	HeuristicCost(x, y graph.Node) float64
 }
 
-// Returns an ordered list consisting of the nodes between start and goal. The path will be the
-// shortest path assuming the function heuristicCost is admissible. The second return value is the
-// cost, and the third is the number of nodes expanded while searching (useful info for tuning
-// heuristics). Negative Costs will cause bad things to happen, as well as negative heuristic
-// estimates.
+// AStar returns the A*-shortest path from s to t in g using the heuristic h, and the
```

Comment:
Although after repeated reading the text seems clear, I would still change to make it clearer what is returned and where. I leave it up to you, but what about `AStar finds the A*-shortest path from s to t in g using the heuristic h. The path and its cost are returned in a Shortest ...`?